soc/realtek/ec/rts5912/debug_swj.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+ * Author: Titan Chen <titan.chen@realtek.com>
+ */
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/init.h>
+
+#define SWJ_NODE DT_NODELABEL(swj_port)
+
+PINCTRL_DT_DEFINE(SWJ_NODE);
+
+const struct pinctrl_dev_config *swj_pcfg = PINCTRL_DT_DEV_CONFIG_GET(SWJ_NODE);
+
+static int swj_connector(void)
+{
+	int err;
+
+	err = pinctrl_apply_state(swj_pcfg, PINCTRL_STATE_DEFAULT);
+	if (err < 0) {
+		__ASSERT(0, "SWJ pinctrl setup failed");
+		return err;
+	}
+
+	return 0;
+}
+
+SYS_INIT(swj_connector, PRE_KERNEL_1, CONFIG_SWJ_CONNECTOR_PRIORITY);
```

Comment:
use soc hooks

Reply:
We added `soc_early_init_hook` to soc.c and called `swj_connector_init` in new commit. Thanks.